hypervisor/src/mshv/x86_64/emulator.rs

```diff
@@ -88,6 +100,36 @@ impl PlatformEmulator for MshvEmulatorContext<'_> {
 
         Ok(())
     }
+}
+
+/// Platform emulation for Hyper-V
+impl PlatformEmulator for MshvEmulatorContext<'_> {
+    type CpuState = EmulatorCpuState;
+
+    fn read_memory(&self, gva: u64, data: &mut [u8]) -> Result<(), PlatformError> {
```

Comment:
Any reason we separate `read_memory()` and `read_memory_flags()`?
I ask because we have the cross -page logic for write_memory just below here, but for read_memory it's off in another function...

Reply:
`read_memory` is a member of the PlatformEmulator trait, while `read_memory_flags` is part of the implementation of `MshvEmualtorContext`. The latter is called by both `read_memory` and `fetch` but with different flags.

It doesn't make sense to introduce a `write_memory_flags` because there is only one place that needs to write to memory.

Reply:
makes sense